gateway/internal/petition/petition_repository.go

```diff
@@ -27,17 +30,83 @@ type petitionRepository struct {
 	client pb.PetitionServiceClient
 }
 
-func (svc *petitionRepository) Get() {
+func mapPetition(pbPetition *pb.Petition) model.Petition {
+	var petition model.Petition
+
+	petition.Title = pbPetition.Title
+	petition.Category = pbPetition.Category
+	petition.Description = pbPetition.Description
+	petition.Image = pbPetition.Image
+	petition.Status = uint32(pbPetition.Status)
+	petition.UserID = uint(pbPetition.UserId)
+
+	return petition
 }
 
-func (svc *petitionRepository) Delete() {
+func (repo *petitionRepository) CreatePetition(petition model.Petition) (uint32, error) {
+	resp, err := repo.client.CreatePetition(context.Background(), &pb.CreatePetitionRequest{
+		Petition: &pb.Petition{
+			PetitionId: uint32(petition.PetitionId),
+			Title:      petition.Title,
+			Category:   petition.Category,
+			Image:      petition.Image,
+			Status:     uint32(petition.Status),
+			UserId:     uint32(petition.UserID),
+		},
+	})
+
+	if err != nil {
+		return 0, err
+	}
+	return resp.PetitionId, nil
+}
+
+func (repo *petitionRepository) GetPetitions(query model.PaginationQuery) ([]model.Petition, error) {
+	resp, err := repo.client.GetPetitions(context.Background(), &pb.GetPetitionsRequest{
+		Page:  query.Page,
+		Limit: query.Limit,
+	})
+
+	var petitions []model.Petition
+
+	if err != nil {
+		return petitions, err
+	}
+
+	for _, pbpetition := range resp.Petitions {
+		petitions = append(petitions, mapPetition(pbpetition))
+	}
 
+	return petitions, nil
 }
 
-func (svc *petitionRepository) Create() {
+func (repo *petitionRepository) UpdatePetition(id uint32, status string) (string, error) {
+	resp, err := repo.client.UpdatePetition(context.Background(), &pb.UpdatePetitionRequest{
+		Id:     id,
+		Status: status,
+	})
+	var message string
+	if err != nil {
+		return message, nil
+	}
+
+	message = resp.Message
+
+	return message, nil
 
 }
 
-func (svc *petitionRepository) Update() {
+func (repo *petitionRepository) DeletePetition(id uint32) (string, error) {
```

Comment:
Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?